### PR TITLE
chore: add release scripts

### PR DESCRIPTION
**Pull request by mkilpatrick**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/cli",
-  "version": "1.0.0",
+  "version": "1.0.0-rc.0",
   "description": "Command-line Interface for the Yext Platform",
   "main": "dist/index.js",
   "author": "jwartofsky@yext.com",
```
Comment (asanehisa) [inline package.json:6]: random, is this the author we wanna keep for this?

File: scripts/release.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  console.error(`invalid target version: ${targetVersion}`);
+  process.exit(1);
+}
+
+const tag = `v${targetVersion}`;
+
+if (targetVersion.includes("rc") && !args.tag) {
+  args.tag = "rc";
+}
+if (targetVersion.includes("beta") && !args.tag) {
```
Comment (asanehisa) [inline scripts/release.ts:57]: idk if we care, but would this be better as an else if? no real diff I think unless we get a really messed up tag

File: scripts/release.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  args.tag = "beta";
+}
+if (targetVersion.includes("alpha") && !args.tag) {
+  args.tag = "alpha";
+}
+
+const { yes }: { yes: boolean } = await prompts({
+  type: "confirm",
+  name: "yes",
+  message: `Releasing ${colors.yellow(tag)} Confirm?`,
```
Comment (asanehisa) [inline scripts/release.ts:67]: maybe add a period after ${colors.yellow(tag)}

File: scripts/release.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+const { stdout } = await run("git", ["diff"], { stdio: "pipe" });
+if (stdout) {
+  step("\nCommitting changes...");
+  await runIfNotDry("git", ["add", "-A"]);
+  await runIfNotDry("git", ["commit", "-m", `release: ${tag}`]);
+  await runIfNotDry("git", ["tag", tag]);
+} else {
+  console.log("No changes to commit.");
```
Comment (asanehisa) [inline scripts/release.ts:99]: \n ?

File: scripts/releaseUtils.ts [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+import minimist from "minimist";
+import { fileURLToPath } from "url";
+
+export const args = minimist(process.argv.slice(2));
+
+export const isDryRun = !!args.dry;
+
+if (isDryRun) {
+  console.log(colors.inverse(colors.yellow(" DRY RUN ")));
+  console.log();
```
Comment (asanehisa) [inline scripts/releaseUtils.ts:21]: curious if this is here on purpose

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@yext/cli",
-  "version": "1.0.0",
+  "version": "1.0.0-rc.0",
```
Comment (jwartofsky-yext) [inline package.json:3]: Should this be rc.0?

File: scripts/releaseUtils.ts [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+        tag
+      )} ${colors.gray(`(${sha.slice(0, 5)})`)}`
+    )
+  );
+  await run(
+    "git",
+    ["--no-pager", "log", `${sha}..HEAD`, "--oneline", "--", `.`],
+    { stdio: "inherit" }
+  );
+  console.log();
```
Comment (asanehisa) [inline scripts/releaseUtils.ts:223]: ?

File: package.json (hunk 6 above)
Comment (mkilpatrick) [reply package.json:3]: What do you mean? It is. You mean without a dash? There should be a dash.

File: scripts/release.ts (hunk 2 above)
Comment (mkilpatrick) [reply scripts/release.ts:57]: This was ripped from pagesjs which was ripped from Vite. I made some slight tweaks to it. The cleanness doesn't matter all that much.

File: scripts/releaseUtils.ts (hunk 5 above)
Comment (mkilpatrick) [reply scripts/releaseUtils.ts:21]: Yes, to add a blank line.

File: scripts/release.ts (hunk 4 above)
Comment (mkilpatrick) [reply scripts/release.ts:99]: Ack

**Review by asanehisa — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (jwartofsky-yext) [reply package.json:3]: I meant should this be 1.0.0-rc.6 or 1.0.0-rc.7 instead of 1.0.0-rc.0

Comment (mkilpatrick) [reply package.json:3]: No, this is a different package. They will never all align.

**Pull request merged**